src/model/index.py

```diff
@@ -28,29 +27,25 @@ def serialize(self):
         os.makedirs(self.__path__, exist_ok=True)
 
     def get_primary_keys(self, value_name):
-        Index.__validate_value_name__(value_name)
-        path = os.path.join(self.__path__, "{}.json".format(value_name))
-        if os.path.exists(path):
-            with open(path, 'r') as file:
-                return json.load(file)
-        else:
-            return []
+        return IndexValue(self, value_name).get_primary_keys()
 
-    @staticmethod
-    def __update_value__(path, value_name, primary_keys):
-        with open(os.path.join(path, "{}.json".format(value_name)), 'w') as file:
-            json.dump(primary_keys, file)
+    def __update_value__(self, value_name, primary_keys):
+        if not primary_keys:
+            pathlib.Path(os.path.join(self.__path__, "{}.json".format(value_name))).unlink()
+        else:
+            with open(os.path.join(self.__path__, "{}.json".format(value_name)), 'w') as file:
+                json.dump(primary_keys, file)
 
     def add_value(self, value_name, primary_key):
-            value = self.get_primary_keys(value_name)
-            value.append(primary_key)
-            self.__update_value__(self.path, value_name, value)
+        primary_keys = self.get_primary_keys(value_name)
+        primary_keys.append(primary_key)
+        self.__update_value__(value_name, primary_keys)
 
     def remove_value(self, value_name, primary_key):
-        value = self.get_primary_keys(value_name)
-        if primary_key in value:
-            value.remove(primary_key)
-        if not value:
-            pathlib.Path(os.path.join(self.path, "{}.json".format(value_name))).unlink()
-        else:
-            self.__update_value__(self.path, value_name, value)
+        primary_keys = self.get_primary_keys(value_name)
+        if primary_key in primary_keys:
+            primary_keys.remove(primary_key)
```

Comment:
why do u need check this. I think remove method do this for free, right?

Reply:
No, it raises an error if it is not found.